### PR TITLE
[SYCL] Fix devicelib identification for NVPTX

### PR DESCRIPTION
**Pull request by frasercrmck**
Recent upstream changes removed the workaround that required libraries to be named ".cubin".

We also renamed the NVPTX libdevice library when we unified it into a single bytecode library.

Unfortunately both of these things broke the driver's brittle logic that determined whether or not an object was a device library. This resulted in us not stripping any symbols from the final object, which is both slower and broke some SYCL CTS tests.

This commit more clearly delineates logic between NVPTX and NativeCPU, as the latter target didn't actually link against anything containing 'libdevice' so was too lenient.

### CODE REVIEW

File: clang/test/Driver/sycl-nvptx-link.cpp [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// REQUIRES: nvptx-registered-target
+
+// RUN: %clang -### -nocudalib -fsycl -fsycl-targets=nvptx64-nvidia-cuda %s 2>&1 | FileCheck %s
```
Comment (mdtoguchi) [inline clang/test/Driver/sycl-nvptx-link.cpp:3]:
```suggestion
// RUN: %clang -### -nocudalib -fsycl -fsycl-targets=nvptx64-nvidia-cuda --sysroot=%S/Inputs/SYCL %s 2>&1 | FileCheck %s
```
Adding the internal sysroot should allow for dropping the `REQUIRES: nvptx-registered-target` dependency.

File: clang/test/Driver/sycl-nvptx-link.cpp [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+// logic the driver uses.
+
+// CHECK: llvm-link
+
+// CHECK: llvm-link
+// CHECK-SAME: -only-needed
+// CHECK-SAME: devicelib--cuda.bc
+// CHECK-SAME: libspirv-nvptx64-nvidia-cuda.bc
+
+void func(){}
```
Comment (mdtoguchi) [inline clang/test/Driver/sycl-nvptx-link.cpp:19]: Source is not needed for a `-###` test.

File: clang/test/Driver/sycl-nvptx-link.cpp (hunk 1 above)
Comment (frasercrmck) [reply clang/test/Driver/sycl-nvptx-link.cpp:3]: Oh I'll give that a go, thanks

**Review by mdtoguchi — APPROVED** [review]
LGTM - thanks!

**Comment by frasercrmck** [comment]
@intel/llvm-gatekeepers this is ready to merge, thanks - there's one unrelated unexpected pass

**Pull request merged**